### PR TITLE
Update keyvault recording path and proxy docs

### PR DESCRIPTION
**Pull request by benbp**
- Update the recording path in keyvault proxy start to the location including the assets.json config
- Update test docs to show correct usage of `recording.StartTestProxy`

### CODE REVIEW

**Review by RickWinter — APPROVED** [review]

File: sdk/internal/recording/README.md [hunk 1]
```diff
@@ -18,11 +18,11 @@ $ENV:PROXY_CERT="C:/ <path-to-repo> /azure-sdk-for-go/eng/common/testproxy/dotne
...
+Recording and playing back tests relies on the [Test Proxy](https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md) to intercept traffic. The recording package can automatically install and run an instance of the test-proxy server per package. The below code needs to be added to test setup and teardown in order to achieve this. The service directory value should correspond to the directory containing the [assets.json](https://github.com/Azure/azure-sdk-tools/blob/main/tools/assets-automation/asset-sync/README.md#assetsjson-discovery) config, see [examples](https://github.com/search?q=repo%3AAzure%2Fazure-sdk-for-go+assets.json+language%3AJSON&type=code&l=JSON).
 
 ```golang
 func TestMain(m *testing.M) {
-	proxy, err := recording.StartTestProxy(nil)
+	proxy, err := recording.StartTestProxy("<path to service directory with assets.json file", nil)
```
Comment (jhendrixMSFT) [inline sdk/internal/recording/README.md:25]: Nit, missing closing `>` in example text.

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by benbp** [comment]
Not sure why the pipelines aren't auto-running here

**Comment by gracewilcox** [comment]
> Not sure why the pipelines aren't auto-running here

Same thing happened with [this pull request](https://github.com/Azure/azure-sdk-for-go/pull/21854). Is there something wrong with the pipelines?

**Review by RickWinter — APPROVED** [review]

**Pull request merged**